packages/common/src/store/buy-crypto/sagas.ts

```diff
@@ -0,0 +1,497 @@
+      )
+    })
+
+    // Create a temp wSOL account
+    const walletSolTokenAccount = getAssociatedTokenAddressSync(
+      NATIVE_MINT,
+      wallet.publicKey
+    )
+    const createWSOLInstruction =
+      createAssociatedTokenAccountIdempotentInstruction(
+        feePayer, // fee payer
+        walletSolTokenAccount, // account to create
+        wallet.publicKey, // owner
+        NATIVE_MINT // mint
+      )
+
+    // Transfer the SOL to the wSOL account
+    const transferWSOLInstruction = SystemProgram.transfer({
+      fromPubkey: wallet.publicKey,
+      toPubkey: walletSolTokenAccount,
+      lamports: BigInt(quoteResponse.otherAmountThreshold)
+    })
+    const syncNativeInstruction = createSyncNativeInstruction(
+      walletSolTokenAccount
+    )
+
+    // Swap the new SOL amount into the desired token amount
+    const { swapInstruction, addressLookupTableAddresses } = yield* call(
+      [jupiterInstance, jupiterInstance.swapInstructionsPost],
+      {
+        swapRequest: {
+          quoteResponse,
+          userPublicKey: wallet.publicKey.toBase58(),
+          destinationTokenAccount: userbank.toBase58(),
+          useSharedAccounts: true
+        }
+      }
+    )
+
+    // Close the temporary wSOL account
+    const closeWSOLInstruction = createCloseAccountInstruction(
+      walletSolTokenAccount, //  account to close
+      feePayer, // fee destination
+      wallet.publicKey //  owner
+    )
+
+    const instructions = [
+      memoInstruction,
+      createWSOLInstruction,
+      transferWSOLInstruction,
+      syncNativeInstruction,
+      parseJupiterInstruction(swapInstruction),
+      closeWSOLInstruction
+    ]
+    const { transaction, addressLookupTableAccounts } = yield* call(
+      createVersionedTransaction,
+      audiusBackendInstance,
+      {
+        instructions,
+        lookupTableAddresses: addressLookupTableAddresses,
+        feePayer
+      }
+    )
+    transaction.sign([wallet])
+
+    const { error } = yield* call(
+      relayVersionedTransaction,
+      audiusBackendInstance,
+      {
+        transaction,
+        addressLookupTableAccounts,
+        skipPreflight: true
+      }
+    )
+
+    if (error) {
+      throw new BuyCryptoError(BuyCryptoErrorCode.SWAP_ERROR, error)
+    }
+
+    // Record success
+    if (mint === 'audio') {
+      yield* call(
+        track,
+        make({
+          eventName: Name.BUY_AUDIO_SUCCESS,
+          provider,
+          requestedAudio: amount,
+          surplusAudio:
+            amount - Number(quoteResponse.outAmount) / outputTokenLamports,
+          actualAudio: Number(quoteResponse.outAmount) / outputTokenLamports
+        })
+      )
+    } else if (mint === 'usdc') {
+      yield* call(
+        track,
+        make({
+          eventName: Name.BUY_USDC_SUCCESS,
+          provider,
+          requestedAmount: amount
+        })
+      )
+    }
+    yield* put(buyCryptoSucceeded())
```

Comment:
maybe else { throw new Error(`Invalid mint ${mint}`) or something

Reply:
I don't think we get this far with invalid mint fwiw. Might make sense though to have up front guard